jstreegrid.js

```diff
@@ -1035,6 +1035,31 @@
 				last = a;
 				// find which column our tree shuld go in
 				var s = this.settings.grid;
+
+				var prevPos = pos - 1;
+
+				while (prevPos >= 0 && this.is_hidden(peers[prevPos])) {
+					prevPos--;
+				}
+
+				gridCellPrevId = prevPos < 0 ? objData.parent : findLastClosedNode(this, peers[prevPos]);
+				var nextPos = pos + 1;
+				gridCellNextId = pos >= peers.length - 1 ? "NULL" : peers[nextPos];
+				while (gridCellNextId !== "NULL" && this.is_hidden(gridCellNextId)) {
+					gridCellNextId = nextPos >= peers.length - 1 ? "NULL" : peers[++nextPos];
+				}
+
+				var childPos = 0;
+				gridCellChildId = "NULL";
```

Comment:
.. and this line (with the appropriate code) should be in its own PR. Let's keep the logic for performance enhancements and logic for bugfixes separate.